libr/core/anal_vt.c

```diff
@@ -52,28 +63,31 @@ static int isVtableStart(RCore *core, ut64 curAddress) {
 	if (!curAddress || curAddress == UT64_MAX) {
 		return false;
 	}
-	if (valueInTextSection (core, curAddress)) {
-		// total xref's to curAddress
-		RList *xrefs = r_anal_xrefs_get (core->anal, curAddress);
-		if (!r_list_empty (xrefs)) {
-			r_list_foreach (xrefs, xrefIter, xref) {
-				// section in which currenct xref lies
-				if (inTextSection (core, xref->addr)) {
-					r_io_read_at (core->io, xref->addr, buf, VTABLE_BUFF_SIZE);
-					if (r_asm_disassemble (core->assembler, &asmop, buf, VTABLE_BUFF_SIZE) > 0) {
-						if ((!strncmp (asmop.buf_asm, "mov", 3)) ||
-						    (!strncmp (asmop.buf_asm, "lea", 3))) {
-							return true;
-						}
-					}
-				}
+	if (!valueInTextSection (core, curAddress)) {
+		return false;
+	}
+	// total xref's to curAddress
+	RList *xrefs = r_anal_xrefs_get (core->anal, curAddress);
+	if (r_list_empty (xrefs)) {
+		return false;
+	}
+	r_list_foreach (xrefs, xrefIter, xref) {
+		// section in which currenct xref lies
+		if (inTextSection (core, xref->addr)) {
+			r_io_read_at (core->io, xref->addr, buf, VTABLE_BUFF_SIZE);
+			if (!r_asm_disassemble (core->assembler, &asmop, buf, VTABLE_BUFF_SIZE) > 0) {
+				continue;
+			}
+			if ((!strncmp (asmop.buf_asm, "mov", 3)) ||
```

Comment:
do this only for x86, also I would use AnalOp instead.

Reply:
Exactly, that would be my next step.